naturescall/views.py

```diff
@@ -388,27 +388,40 @@ def delete_rating(request, r_id):
 
 # The page for adding new restroom to our database
 @login_required(login_url="login")
-def add_restroom(request, r_id):
+def add_restroom(request, yelp_id):
+    # check to see if restaurant already exists in database
+    current_restroom_set = Restroom.objects.filter(yelp_id=yelp_id)
     if request.method == "POST":
-        f = AddRestroom(request.POST)
-        if f.is_valid():
-            post = f.save(commit=False)
-            post.save()
+        form = AddRestroom(request.POST)
+        if form.is_valid():
+            entry = form.save(commit=False)
+            if current_restroom_set:
+                entry.id = current_restroom_set[0].id
+                entry.save()
+                return redirect(
+                    "naturescall:manage_restroom", r_id=current_restroom_set[0].id
+                )
+            entry.save()
             return HttpResponseRedirect(reverse("naturescall:index"))
         else:
-            return render(request, "naturescall/add_restroom.html", {"form": f})
+            return render(request, "naturescall/add_restroom.html", {"form": form})
     else:
-        k = get_business(api_key, r_id)
-        if k.get("error"):
-            raise Http404("Restroom does not exist")
-        context = {}
-        name = k["name"]
-        title = (
-            k["name"] + " " + k["location"]["address1"] + " " + k["location"]["city"]
-        )
-        form = AddRestroom(initial={"yelp_id": r_id, "title": title})
-        context["form"] = form
-        context["name"] = name
+        if not current_restroom_set:
```

Comment:
I reviewed the PR, all looks good except for this part I don't quite understand what's going on here. I don't get why we need to check if the new restroom added is already in our database. Is this trying to solve the issue of multiple user concurrently adding the same location at the same time?